File: warmup/sherlock-and-the-beast/code.cpp

#include <cstdio>  
#include <cstring>  
#include <iostream>  
#include <algorithm>  
#include <map>  

using namespace std;

int main()  
{      
    
    int cases, digits, fives, threes, remainder;

    cin >> cases;

    while (cases--)  
    {  
	
	cin >> digits;	
	
	if(digits < 9){
	    
	    switch(digits){
		
		case 3 : { cout << "555"; break;}
		case 5 : { cout << "33333"; break;}
		case 6 : { cout << "555555"; break;}
		case 8 : { cout << "55533333"; break;}
		default: { cout << "-1";}
	    }
	    
	}
	else {
	
		remainder = digits - 9;
		
		switch(remainder % 3){

		    case 0 : { threes = 0; break;}
		    case 1 : { threes = 10; break;}
		    case 2 : { threes = 5; break;}
		}
		
		fives = digits - threes;

		for(int i = 0; i < digits; i++){

		    if(i < fives){

			cout << 5;
		    }
		    else {
			cout << 3;
		    }
		} 
	}
	
	cout << endl; 
	
    }  
    
    return 0;  
} 

File: warmup/sherlock-and-the-beast/python/code.py

import sys

cases = input()

for i in range(0, cases):
    
    digits = input()

    # This problem follows a simple numeric progression. Just write
    # out the series of numbers and the relationships are easy to spot
    
    if digits < 9:
	
	if digits == 3:
	    sys.stdout.write("555")
	    
	elif digits == 5:
	    sys.stdout.write("33333")
	    
	elif digits == 6:
	    sys.stdout.write("555555")
	    
	elif digits == 8:	    
	    sys.stdout.write("55533333")
	    
	else:
	    sys.stdout.write("-1")
	    
    else:
	
	remainder = digits - 9
	
	if remainder % 3 == 0:
	    threes = 0
	    
	elif remainder % 3 == 1:
	    threes = 10
	    
	elif remainder % 3 == 2:
	    threes = 5
	    
	fives = digits - threes
	
	for i in range(0,digits):
	    
	    if i < fives:		
		sys.stdout.write("5")	
		
	    else:
		sys.stdout.write("3")
		
    sys.stdout.write("\n")